### PR TITLE
deprecate Python2 support

### PR DESCRIPTION
**Pull request by bcwu**
### Description

deprecate Python2 support

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -16,7 +16,7 @@ project_urls =
 
 [options]
 install_requires =
-    rsconnect-python>=1.5.4
+    rsconnect-python>=1.7.0
```
Comment (mmarchetti) [inline setup.cfg:19]: let's make this 1.7.1

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        python-version: ['2.7', '3.5', '3.6', '3.7', '3.8', '3.9']
+        python-version: ['3.5', '3.6', '3.7', '3.8', '3.9']
```
Comment (mmarchetti) [inline .github/workflows/main.yml:17]: should we add 3.10?

File: setup.cfg [hunk 3]
```diff
@@ -25,6 +25,6 @@ install_requires =
 setup_requires =
     setuptools
 packages = rsconnect_jupyter
-python_requires = >=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*
+python_requires = !=2.*, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*
```
Comment (mmarchetti) [inline setup.cfg:28]: This matches rsconnect-python so is OK. Could also consider making this `>=3.5`; the more complex pattern was needed previously to give us a discontiguous range.

**Review by mmarchetti — APPROVED** [review]

**Comment by bcwu** [comment]
fixes #20780

**Comment by bcwu** [comment]
fixes https://github.com/rstudio/connect/issues/20852

**Pull request merged**